runtime/src/bank.rs

```diff
@@ -17423,7 +17430,23 @@ pub(crate) mod tests {
             &validator_keypairs,
             vec![LAMPORTS_PER_SOL; 2],
         );
-        let bank = Arc::new(Bank::new_for_tests(&genesis_config));
+        let bank = Arc::new(Bank::new_with_paths(
+            &genesis_config,
+            Arc::<RuntimeConfig>::default(),
+            Vec::new(),
+            None,
+            None,
+            AccountSecondaryIndexes::default(),
+            AccountShrinkThreshold::default(),
+            false,
+            Some(AccountsDbConfig {
+                // at least one tests hit this assert, so disable it
```

Comment:
Does this need a `#[should_panic]` test?
https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html#testing-panics

Reply:
I am not sure whether the test that panics SHOULD panic or not. That is something I hope you review.

Reply:
well, there shouldn't be any tests panicking right now,
unless we intentionally create a stakes-cache inconsistent with accounts-db.

Reply:
@behzadnouri 
This test modifies the owner. When the assert is enabled (as it usually is by default for testing), this test fails because of the results of this code.
test:
test_stake_vote_account_validity
```
        // Modify staked vote account owner; a vote account owned by another program could be
        // freely modified with malicious data
        let bogus_vote_program = Pubkey::new_unique();
        vote_account.set_lamports(original_lamports);
        vote_account.set_owner(bogus_vote_program);
        bank.store_account(
            &validator_vote_keypairs0.vote_keypair.pubkey(),
            &vote_account,
        );
        ```
        
        changing this to true causes the test to fail:
        ```
           assert_stakes_cache_consistency: false,
           ```
in:
```
fn check_stake_vote_account_validity<F>(
```
So, this is a test that IS failing when I enable the assert. Are you familiar with this test? Obviously I added the assert. Thankfully it does not go off on any other test.

Reply:
yeah, the comment here:
https://github.com/solana-labs/solana/blob/a9287ab4c/runtime/src/bank.rs#L17403-L17405
is describing what is happening; also see:
https://discord.com/channels/428295358100013066/439194979856809985/964210544397660200
Current stakes-cache interceptor code assumes that a stake or vote account owner cannot change from respectively `solana_{stake,vote}_program` id. This is true today and we have to preserve that going forward.

So I think might make sense to adapt that test to create a `#[should_panic]` test when `assert_stakes_cache_consistency` is true; (as a separate test, not replacing the current `test_stake_vote_account_validity`).